Sources/SupabaseStorage/StorageBucketApi.swift

```diff
@@ -10,93 +10,167 @@ public class StorageBucketApi: StorageApi {
+
+    let responseData = try await fetch(
       url: url,
       method: .post,
-      parameters: ["id": id, "name": id],
+      json: body,
       headers: headers
     )
-    guard let dict = response as? [String: Any] else {
-      throw StorageError(message: "failed to parse response")
+
+    let responseObject = try decoder.decode(OneKeyedResponse<String>.self, from: responseData)
+
+    if responseObject.key != "name" {
+      throw StorageError(message: "failed to parse response - missing 'name'")
     }
-    return dict
+
+    return responseObject.value
   }
 
-  /// Removes all objects inside a single bucket.
+  /// Updates a Storage bucket
+  ///
+  /// > A bucket with `id` must already exist
+  ///
+  /// > RLS policy permissions required:
+  /// > - `buckets` table permissions: `update`
+  /// > - `objects` table permissions: none
+  ///
   /// - Parameters:
-  ///   - id: The unique identifier of the bucket you would like to empty.
-  public func emptyBucket(id: String) async throws -> [String: Any] {
-    guard let url = URL(string: "\(url)/bucket/\(id)/empty") else {
-      throw StorageError(message: "badURL")
+  ///   - isPublic: The visibility of the bucket.
+  /// - Returns: newly created bucket ID
+  public func updateBucket(id: String, isPublic: Bool = false) async throws {
+    let url = url.appendingPathComponent("/bucket/\(id)")
+
+    let body = Bucket.CreateRequest(id: id, isPublic: isPublic)
+
+    let responseData = try await fetch(
+      url: url,
+      method: .put,
+      json: body,
+      headers: headers
+    )
+
+    let responseObject = try decoder.decode(OneKeyedResponse<String>.self, from: responseData)
+
+    guard responseObject.key == "message", responseObject.value == "Successfully updated" else {
+      throw StorageError(message: "'\(responseObject.key)' = '\(responseObject.value)'")
     }
+  }
 
-    let response = try await fetch(url: url, method: .post, parameters: [:], headers: headers)
-    guard let dict = response as? [String: Any] else {
-      throw StorageError(message: "failed to parse response")
+  /// Removes all objects inside a single bucket.
+  ///
+  /// > RLS policy permissions required:
+  /// > - `buckets` table permissions: `select`
+  /// > - `objects` table permissions: `select` and `delete`
+  ///
+  /// - Parameter id: The unique identifier of the bucket you would like to empty.
+  public func emptyBucket(id: String) async throws {
+    let url = url.appendingPathComponent("/bucket/\(id)/empty")
+
+    let responseData = try await fetch(
+      url: url,
+      method: .post
+    )
+
+    let responseObject = try decoder.decode(OneKeyedResponse<String>.self, from: responseData)
+
+    if responseObject.key != "message" {
+      throw StorageError(message: "failed to parse response - missing 'message'")
+    }
+
+    if responseObject.value != "Successfully emptied" {
+      throw StorageError(message: "response error: \(responseObject.value)")
```

Comment:
Relying on string comparison is always a bad idea, you could actually use status code for deciding this.